services/packages/cargo/index.go

```diff
@@ -107,9 +107,16 @@ func RebuildIndex(ctx context.Context, doer, owner *user_model.User) error {
 }
 
 func AddOrUpdatePackageIndex(ctx context.Context, doer, owner *user_model.User, packageID int64) error {
-	repo, err := getOrCreateIndexRepository(ctx, doer, owner)
+	// We do not want to force the creation of the repo here
+	// cargo http index does not rely on the repo itself,
+	// so if the repo does not exist, we just do nothing.
+	// This partially resolves #26844 (error 500 when trying to publish a crate if user is missing write access to the repo)
```

Comment:
```suggestion
```
I would remove this line because it's not really a solution. If the repo is used and the user has no write access, the repo is out of sync with the packages in the database afterwards.

Reply:
I think what happens is: If the repo is used, the package is removed from gitea package & a 500 error is returned.